framework/build.xml

```diff
@@ -362,6 +368,52 @@
         </antcall>
     </target>
 
+    <target name="signature-inference-tests" depends="jar,build-tests"
+            description="Run tests for signature inference using .jaif files">
+        <delete dir="build/signature-inference"/>
+        <!--Runs the checker on test files with the signature inference on
+             to generate .jaif files.-->
+        <antcall target="-run-tests">
+            <param name="param" value="tests.SignatureInferenceTest"/>
+        </antcall>
+
+        <delete dir="tests/signature-inference/annotated"/>
+
+        <!-- Copying all test files to another directory, removing all
+             expected errors that should not occur after inserting inferred
+             annotations from .jaif files. -->
+        <copy todir="tests/signature-inference/annotated">
+            <fileset dir="tests/signature-inference/non-annotated"/>
+            <filterchain>
+                <linecontains negate="true">
+                    <contains value="//:: error:"/>
+                </linecontains>
+            </filterchain>
+        </copy>
+
+        <!-- The only file that expected errors are maintained is
+             ExpectedErrors.java, so we copy it over. -->
+        <delete file="tests/signature-inference/annotated/ExpectedErrors.java"/>
+        <!-- Inserting annotations from .jaif files in-place. -->
+        <apply executable="insert-annotations-to-source" parallel="true">
+            <srcfile/>
+            <fileset dir="build/signature-inference/" includes="*.jaif"/>
+            <fileset dir="tests/signature-inference/annotated/" includes="*.java"/>
+            <arg value="--outdir=tests/signature-inference/annotated/"/>
+        </apply>
+
+        <copy file="tests/signature-inference/non-annotated/ExpectedErrors.java"
+            todir="tests/signature-inference/annotated"/>
+
+        <!-- Running the checker again on the final .java files. -->
+        <antcall target="-run-tests">
+            <param name="param" value="tests.SignatureInferenceValidationTest"/>
+        </antcall>
+
+        <delete dir="tests/signature-inference/annotated"/>
```

Comment:
Hmm. If a test does not behave as expected, wouldn't we want to look at this directory?